client/packages/programs/src/JsonForms/components/DateOfBirth.tsx

```diff
@@ -101,21 +100,24 @@ const UIComponent = (props: ControlProps) => {
               },
             }}
           />
-          <Box
-            flex={0}
-            style={{ textAlign: 'end' }}
-            flexBasis={FORM_LABEL_COLUMN_WIDTH}
-          >
+
+          <Box flex={0} style={{ textAlign: 'end' }}>
             <FormLabel sx={{ fontWeight: 'bold' }}>{t('label.age')}:</FormLabel>
           </Box>
-          <Box flex={0}>
-            <NumericTextInput
-              value={age}
-              sx={{ width: 65 }}
-              onChange={onChangeAge}
-              disabled={!props.enabled}
-            />
-          </Box>
+          {(age ?? 1 >= 1) ? (
+            <Box flex={0}>
+              <NumericTextInput
+                value={age}
+                sx={{ width: 65 }}
+                onChange={onChangeAge}
+                disabled={!props.enabled}
+              />
+            </Box>
+          ) : (
+            <Typography fontSize="85%" whiteSpace="nowrap">
+              {t('label.age-months-days', { months, days })}
```

Comment:
I think it's worth hiding months or days if they are 0. Which also means using translations string individually rather then age-months-days

Reply:
> I think it's worth hiding months or days if they are 0.

Yeah, makes sense to hide months if it's 0, though I'm less sure about 0 days. If you read "3 months" you kind of assume it's been rounded from within 2.5 - 3.5 months -- you need the 0 days to make it clear that we're being more precise.